internal/shims/shims.go

```diff
@@ -93,7 +93,9 @@ func FindExecutable(conf config.Config, shimName, currentDirectory string) (stri
 				}
 
 				versions.Versions = tempVersions
-				existingPluginToolVersions[plugin] = versions
+				if len(versions.Versions) > 0 {
+					existingPluginToolVersions[plugin] = versions
+				}
```

Comment:
The code was setting a tool and version even when `tempVersions` was empty, which was wrong.